### PR TITLE
feat: add fasthtml demo-app

### PR DESCRIPTION
**Pull request by thomasht86**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Adds a sample app with a fastHTML-frontend. 
All details should be in the README.

Deployed to https://huggingface.co/spaces/vespa-engine/fasthtml-vespa 
Slack-msg if you need access. Intention is to make it public after merge. 

The real test would be if someone else tried to deploy by following the README.

@lperry25 / @ldalves : Feel free to suggest some UI changes, if you see something that could be improved without adding too much complexity. 
cc: @kkraune @jobergum

### CODE REVIEW

**Comment by thomasht86** [comment]
Guess the space needs to be publicly available for the link check to pass, so did that now.

**Comment by jobergum** [comment]
Great! 

Minor comments:

- Provide search examples to understand what the system is capable of 
- Make hitting Enter fire the query 
- It would be great to expose some settings like rank profile etc to illustrate that this is not a one default search/ranking to rule them all.

**Review by lperry25 — COMMENTED** [review]
I think it looks mostly great. Just one small UI feedback point :)

File: examples/fasthtml-demo/main.py [hunk 1]
```diff
@@ -0,0 +1,721 @@
...
+        Query(query=query, sess_id=sesskey, timestamp=int(time.time()))
+    )
+
+
+def parse_results(records):
+    return [
+        Article(
+            Header(
+                H2(
+                    A(
```
[the remark this answers is not in the document]
Comment (thomasht86) [reply examples/fasthtml-demo/main.py:453]: Hmm, it should open the document when clicked. 
It does not show the actual links in browser though, as it is running in hf.

**Comment by thomasht86** [comment]
Added sample queries and radio button to select rank profile.

**Comment by jobergum** [comment]
Sorry for being picky about this PR :)

image

- When I pick sample query, it fires the query, but the form text input stays the same 
- When a click radio button nothing happens, it seems odd to press search (for me)

**Comment by thomasht86** [comment]
If you're picky it means you care, so no worries :) 
- Didn't think about sample query updating form input. Will fix!
-  About radio button, you think it should fire query on radio button selection? Isn't it more natural to hit Enter or "Search" after selection? Can see if it is simple to do.

File: examples/fasthtml-demo/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lperry25) [reply examples/fasthtml-demo/main.py:453]: Oh I see now I am getting an internal server error when I try to access the links so nothing happens in the UI. This link gives a 500. https://vespa-engine-fasthtml-vespa.hf.space/document/MED-2434

Am I just missing permissions?

[the remark this answers is not in the document]
Comment (thomasht86) [reply examples/fasthtml-demo/main.py:453]: Hmm, should not require permissions, but I got the same from my mobile, when not logged in to hf. Will investigate!

**Comment by ldalves** [comment]
For a demo, I think the overall design is good. :) 

Fixing all the points above, I think it would be nice to also focus the user on the search field when the page loads.

**Comment by thomasht86** [comment]
Fixed filling of sample queries. 
Did you also check out the "admin" login to display the query logs? (admin, admin)
Could be expanded to include results, clicks and maybe LLM-judgments for easy eval bootstrapping.

**Comment by kkraune** [comment]
should we just merge and interate on this?

**Comment by thomasht86** [comment]
I paused the live huggingface-demo, so I can remove the link pointing to that in the README first at least. 
The next iteration could be replacing cord19 UI, but not sure when I will get around to that.

**Comment by thomasht86** [comment]
Removed the link now. If noone else objects, I think we can merge.

**Pull request merged**